torch/utils/cpp_extension.py

```diff
@@ -1908,6 +1970,25 @@ def _prepare_ldflags(extra_ldflags, with_cuda, verbose, is_standalone):
         elif IS_HIP_EXTENSION:
             extra_ldflags.append(f'-L{_join_rocm_home("lib")}')
             extra_ldflags.append('-lamdhip64')
+
+    elif IS_XPU_EXTENSION:
+        library_dirs = library_paths(cuda=False, xpu=True)
+        # Append oneMKL link parameters, detailed please reference:
+        # https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl-link-line-advisor.html
+        oneapi_link_args = []
+        oneapi_link_args += [f"-L{x}" for x in library_dirs]
+        # oneapi_link_args += ['-fsycl-device-code-split=per_kernel']
+        oneapi_link_args += ["-Wl,--start-group"]
+        oneapi_link_args += [f"{x}" for x in _get_one_api_help().get_onemkl_libraries()]
```

Comment:
In my opinion, here should not link `mkl` and `dnnl` by default. We should only link the additional library `libc10_xpu.so` and `libtorch_xpu.so`. For the other library, let the user link by himself via pass them to `extra_compile_args` and `extra_link_args`.
@gujinghui any idea?